Change icons to Lucide Icons

MokkaCicc opened the pull request:
I always found the icons of the linter plugins out of place within Obsidian.
![image](https://github.com/platers/obsidian-linter/assets/38502090/c98b42e0-9581-42e6-8239-038380e9cbf5)

## Changes
Since Obsidian uses the [Lucide Icons pack](https://lucide.dev/), I changed the plugin icons to match the default ones. The column `Alternatives` are other icons I considered, if you think one is better than the other, I can change it!

| Name | Old Icon | New Lucide Icon | Alternatives |
|--------|-----------|--------------------|--------------|
| `lintFileSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/217751ee-6a66-4608-ac13-0a6be1ca099a) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/2158f8fe-4e69-4fdc-8362-a7992cf19c3a) | / |
| `lintFolderSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/0b858567-954f-4eca-943f-c4f485397bb0) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/fd857ff4-f177-442d-aec8-2782f096b7ec) | / |
| `lintVaultSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/a4aff6ae-a0c3-4dd3-ba71-0f95bc8621f5) | ![icon)](https://github.com/platers/obsidian-linter/assets/38502090/5f54747e-5240-433c-aaee-939e79147080) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/dd8fced1-b12a-4863-9328-eade51c2e1c3)[^1] |
| `whitespaceSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/ec1eeda8-efb6-4554-a538-430eee3055c2) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/139198b6-b1ac-4713-9c80-222cc32999c6) | / |
| `mathSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/8332fc3f-810f-4b94-9d4a-3b86e47f792a) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/fbdf42d6-4bb5-4311-b5fa-7ffb83be59ca) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/c04d4aaa-6b8b-4844-b21a-7ecbd64c27cb) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/928d78a1-0c07-4ed8-bf6e-f3432d238994) |
| `contentSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/fc5e892f-4241-47e7-84a5-84cb7e873d1b) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/58c2e2da-0707-4488-a351-3d3114aee302) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/01e0e570-b151-443d-8702-8260b105a004) |
| `pasteSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/f751a84c-620e-4a27-9781-233bd0efcf1b) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/48b7f154-d7e4-4c70-a7f9-7d4d7dfdefed)[^2] | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/a9fb2381-927a-4972-8593-4c60fdb42d5b)[^3] ![icon](https://github.com/platers/obsidian-linter/assets/38502090/de7eb1d3-a7af-4fb0-ba91-b1ab6d6812de) |
| `addBookSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/a79f6c8b-4864-4680-bd73-4e17b1d285dc) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/d1440ba8-a9d8-47c0-8407-a6638dcf7603) | / |
| `headingSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/33ebb0d8-52de-49d0-8f9f-21f5bd4cbe23) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/215c65a0-c4c5-465a-869f-075bd7198096) | / |
| `footerSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/15808826-3c75-47fa-bfab-f7a5bdc848a9) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/59ca5ebe-cdb2-44b3-8e3e-43f3d2726c76) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/16bc3bae-4467-41d3-82c2-5549bf6266c1)[^4] ![icon](https://github.com/platers/obsidian-linter/assets/38502090/eee85690-405d-4f7d-9f82-0171c83b7f35) |
| `yamlSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/44c58bf7-8b59-4362-896a-e1572ce367e5) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/2656b998-2228-4e35-b38f-b4feae60cd24) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/63760476-8cab-4f77-b2e1-4a516f3da4cf)[^5] ![icon](https://github.com/platers/obsidian-linter/assets/38502090/9b51937a-f423-48ee-9fde-de79bfde5715)[^6] ![icon](https://github.com/platers/obsidian-linter/assets/38502090/bcade499-01ec-495a-b53e-59542f0b1353) ![icon](https://github.com/platers/obsidian-linter/assets/38502090/22380fc6-4716-4bde-852d-5a15392a5086) |
| `settingsSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/5a7baecf-fead-459b-a4de-833ac70f19bf) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/471e93d9-967d-499c-a7ea-534a2f72d8ce)[^7] | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/98876def-64cd-4075-8755-d49c0dc07565) ![icon](https://github.com/platers/obsidian-linter/assets/38502090/af1b81bb-bf76-4db1-bacd-9096e1acf2c3) ![icon](https://github.com/platers/obsidian-linter/assets/38502090/b86816bc-3c7f-4ef7-ad4e-c0be9cf693bd) |
| `debugSVG`| ![icon](https://github.com/platers/obsidian-linter/assets/38502090/2415f9f4-b28f-4062-bae7-7ab2837df0bb) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/c2a5f169-4271-47c9-8ebd-5376bfffd42d) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/8ca8d14f-8035-4897-b28c-e864c3bb96a3) |
| `clipboardSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/ef0667c1-bad8-4dd5-8b58-6716fd0660c4) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/8a8c0b32-c46d-4a56-a844-1022401ee94d) | / |
| `successSVG` | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/cb3e3e0b-1531-40e0-bd56-e21e99033a9b) | ![icon](https://github.com/platers/obsidian-linter/assets/38502090/ed8fd222-7ecb-43b1-a286-3ba7bd2e520a) | / |

[^1] This is a brain, the vault being a second brain, this can be fitting.
[^2] Default Obsidian icon for "Paste"
[^3] Default Obsidian icon for "Paste as plain text"
[^4] Default Obsidian icon for links, but not necessarily footnotes
[^5] Default Obsidian icon for the new properties pane
[^6] Default Obsidian icon for the new global properties pane
[^7] Default Obsidian icon for the settings

## Results
![image](https://github.com/platers/obsidian-linter/assets/38502090/0dcfdf69-83b1-46c2-9e2e-45fd913c9ed8)
![image](https://github.com/platers/obsidian-linter/assets/38502090/23a11db2-8b77-457d-bd9a-147d6b3acd84)

MokkaCicc commented:
I mismatched some icon in my PR, I will fix it asap.

pjkaufman commented:
The only icon I do not like the replacement for is the footnotes. I will look at the lucide icons and see if they have a superscript icon.

pjkaufman commented:
Is there a particular reason not to use https://lucide.dev/icons/superscript (i.e x<sup>2</sup>) for the icon for footnotes? I know it could be misconstrued as math, but I think it works better than footprints.

MokkaCicc commented:
> Is there a particular reason not to use https://lucide.dev/icons/superscript (i.e x2) for the icon for footnotes? I know it could be misconstrued as math, but I think it works better than footprints.

I didn't found this icon! I updated the PR. I also added the missing line for the `pasteSVG` in the table.

pjkaufman commented:
This looks good to me. I will go ahead and merge it. Thanks for putting in the effort to add these new icons!
